### PR TITLE
Update Azure Data Explorer query runner to latest version, add managed identity

### PR DESCRIPTION
**Pull request by spacentropy**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description

Hello. Here's the list of things that were improved:
* upgraded Azure Data Explorer(Kusto) library to the latest version
* added Materialized Views to schema viewer
* added Managed Identities support for secrets-less auth
* added query user annotation in request description
* added data_scanned
* added simple test to get_schema

### CODE REVIEW

File: requirements_all_ds.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ phoenixdb==0.7
 # certifi is needed to support MongoDB and SSL:
 certifi>=2019.9.11
 pydgraph==2.0.2
-azure-kusto-data==0.0.35
+azure-kusto-data==2.1.1
```
Comment (susodapop) [inline requirements_all_ds.txt:34]: This is a significant update. Were there any breaking changes between v0 and v2.1.1?

**Review by arikfr — COMMENTED** [review]
Thank you for the PR! How about we split the query runner into two: one for MSI and one for without? This way it will be clear which properties are required for each variant.

**Comment by spacentropy** [comment]
@arikfr sure. Should it be two files or just one? how do I avoid duplicating logic?

**Comment by guidopetri** [comment]
@spacentropy , thanks for the PR! We've updated a lot of things now that we're Community-driven so - if you're still interested in getting this merged - would you mind rebasing off master to re-run the CI?

We're trying to clean up our PR todo list, so if you're not interested, that's fine - we'll close the PR in about a week if we don't hear back. If you're interested in reopening the PR afterwards, we would also very much welcome that.

**Comment by spacentropy** [comment]
Hello! Yes, I will prepare everything this week.

**Comment by guidopetri** [comment]
@spacentropy no worries, take your time. We just wanted to make sure this wasn't an abandoned PR - if it takes a month for you to get around to it it's also fine :)

**Comment by spacentropy** [comment]
@justinclift hi. I'm finally(lol) ready to merge this.

**Comment by spacentropy** [comment]
@eradman @arikfr Hello. I would like to merge this :-) We've been running this branch for a while.

**Pull request closed without merging**